src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/IfError.txt

```diff
@@ -195,3 +198,24 @@ Time(11,30,59,0)
 // Error-DateTime
 >> IfError(Error({Kind: ErrorKind.Validation}), DateTime(2021, 12, 12, 17, 30, 0))
 DateTime(2021,12,12,17,30,0,0)
+
+>> IfError(1, {x:1}, {x:1}, {y:1}, 1)
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
+// IfError can have mismatched arg types, but result can't be used.
+>> IfError(1/0, 1, {x:4})
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
+// IfError accepts the void value. But then result of the expression can't be used. If(true, 1, {x:4}) => void value.
+>> IfError(1/0, 1, If(true, 1, {x:4}), 1)
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
+// Errors are still returned.
+>> IfError(1, 2, 3/0, If(true, 1/0, {x:4}))
+Error({Kind:ErrorKind.Div0})
+
```

Comment:
It would be good to have a similar test where the error is not returned:

    >> IfError(1, 2, 3/0, If(false, 1/0, {x:4}))
    If(true, {test:1}, "Mismatched...")

Reply:
Added!